formatter/formatter.Dockerfile

```diff
@@ -16,15 +16,45 @@ RUN apk update && apk add --no-cache --update \
 
 RUN pip install yapf
 
-COPY formatter/package.json /package.json
-COPY formatter/package-lock.json /package-lock.json
 COPY .prettier* /
-COPY .estlint* /
 COPY .editorconfig* /
 
+# Below we pre-install nodejs depdendencies for various
+# TS codebases we have. We need all dependencies in order to
+# run type-based checks with eslint. For each directory that
+# contains package.json we run npm install or yarn install and
+# save resulted `node_modules` directory as volume.
+
+# Fetch node js dependencies for `formatter` directory.
+ENV FORMATTER_DIR /code/formatter
+RUN mkdir -p $FORMATTER_DIR
+COPY formatter/package.json $FORMATTER_DIR
+COPY formatter/package-lock.json $FORMATTER_DIR
+WORKDIR $FORMATTER_DIR
 RUN npm install
 
-COPY formatter/fmt /fmt
-VOLUME /code
+# Fetch node js dependencies for `browser-test` directory.
+ENV BROWSER_TEST_DIR /code/browser-test
+RUN mkdir -p $BROWSER_TEST_DIR
+COPY browser-test/package.json $BROWSER_TEST_DIR
+COPY browser-test/yarn.lock $BROWSER_TEST_DIR
+WORKDIR $BROWSER_TEST_DIR
+RUN npm install --global yarn
```

Comment:
This `npm install --global ` should go at the top, so it gets cached in a layer that doesn't change when the package.json files change

Reply:
Done.